### PR TITLE
Fix Stripe disable modal text

### PR DESCRIPTION
**Pull request by Mayisha**
When Link is enabled, it is present in the `enabled_payment_method_ids` in the [response](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/admin/class-wc-rest-stripe-settings-controller.php#L228) to the `settings` data call. [This function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-helper.php#L757-L761) returns the data which are used in the Stripe settings page.

As a side effect of Link being present in the `enabled_payment_method_ids` list, the [disable confirmation modal](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js) shows wrong text.

## Changes proposed in this Pull Request:
- Excluding link from `enabled_payment_method_ids` as an express checkout method.

## Testing instructions
- Enable Stripe.
- Enable ECE feature flag (return `true` from the `is_stripe_ece_enabled` function or set the `_wcstripe_feature_ece` option as `yes`)
- Disable all payment methods except Google/Apple Pay and Link.
- Go to **WooCommcerce > Settings > Payments**
- Toggle Stripe to disable the gateway. You should see a confirmation modal.
- In the `develop` branch, the modal shows [the message](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js#L26) intended for the scenario when other payment methods are enabled.

Screenshot 2024-10-21 at 2 34 04 PM

- In this branch, the modal should display [the message](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js#L22) intended for the scenario when only express methods are enabled.
Screenshot 2024-10-21 at 3 30 19 PM

- Test Link payment method to make sure there is no regression.

### CODE REVIEW

**Review by annemirasol — CHANGES_REQUESTED** [review]
Found this odd behavior:

1. Check out this branch.
2. Go to Payment Methods and enable Link.
3. Refresh the page.
4. Notice that Link is unchecked.
5. Checkout `develop` and refresh the page.
6. Notice that Link is checked.

File: client/blocks/express-checkout/index.js [hunk 1]
```diff
@@ -57,6 +60,7 @@ const expressCheckoutElementsApplePay = ( api ) => ( {
 		} );
 	},
 	paymentMethodId: PAYMENT_METHOD_EXPRESS_CHECKOUT_ELEMENT,
+	gatewayId: 'stripe',
```
Comment (Mayisha) [inline client/blocks/express-checkout/index.js:63]: The changes in this file are not relevant to the modal text, also it does not visibly do anything. I just piggybacked these little changes here. These attributes were missing from the object.

**Comment by Mayisha** [comment]
> Found this odd behavior

Thanks @annemirasol. That's a good finding which I totally missed.

I moved the filtering to the modal specific file in the frontend [93a647a](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3526/commits/93a647afcb11e70e21a4ead6218feb31ed90b3d7), so it should not create regression anywhere else now

**Review by annemirasol — APPROVED** [review]
Thanks for fixing ✨ Code looks good!

✅ I tested turning off the Stripe plugin with different methods enabled, and everything looks good.
✅ Smoke-tested express payment purchases

**Comment by annemirasol** [comment]
@Mayisha Forgot to mention I noticed the PR doesn't have changelog/readme entries. Maybe it can piggyback on your next PR? Or perhaps not even necessary if we're switching to the Jetpack changelogger soon. FYI anyways.

**Comment by Mayisha** [comment]
Thank @annemirasol. I prepared the changelog commit locally but forgot to push the commit 😳

**Pull request merged**